### DISCUSSION TITLE
Validator blocks whole UI when from has many child components

### DISCUSSION THREAD

**Discussion by joer33304** [Blazored/FluentValidation#134] [Q&A]
I do have a form which has controls which have controls resulting in possible hundreds of controls.

The standard blazor validation works fine but when I use the FluentValidator with RuleForEach (of the items rendered by the sub controls) the problem seems to be that even if only one sub item changed, the whole tree is evaluated. The UI hangs for seconds while doing that.

I tried to add a validator at the sub control, but this does not even get instantiated even though I registered it via DI.
So what is the best solution for this ?

**Comment by Sharaf-Mansour** [comment]
Can you show me an example of your rules.
Because I might have an idea for your scenario.
